features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Endpoints/Endpoints.jsx

```diff
@@ -88,8 +89,9 @@ function Endpoints(props) {
 
     useEffect(() => {
         const { lifeCycleStatus } = api;
-        const implType = api.endpointConfig.implementation_status;
-        setModifiedAPI(cloneDeep(api));
+        const apiClone = cloneDeep(api.toJSON());
```

Comment:
The reason to make a deep copy of the api is to avoid updating the original api in the context. We can copy only the required set but when we add those back to the api, the same issue occurs.